test/unit/resources/aws_ec2_instances_test.rb

```diff
@@ -0,0 +1,191 @@
+require 'helper'
+
+class TestAwsEc2Instances < Minitest::Test
+
+#=============================================================================#
```

Comment:
So, this obviously uses a different style that the rest of AWS unit tests - which is OK in general - but we really do have to avoid dependency injection in the constructor.  If you rework this to use the BackendFactory and place your fixture data in a dummy backend, that would be fine.  Have a look at the other AWS plural resources - aws_iam_policies or aws_security_groups would be typical.